src/subprocess-posix.cc

```diff
@@ -262,17 +322,30 @@ bool SubprocessSet::DoWork() {
     fds.push_back(pfd);
     ++nfds;
   }
+  if (nfds == 0) {
+    // Add a dummy entry to prevent using an empty pollfd vector.
+    // ppoll() allows to do this by setting fd < 0.
+    pollfd pfd = { -1, 0, 0 };
+    fds.push_back(pfd);
+    ++nfds;
+  }
 
   interrupted_ = 0;
+  s_sigchld_received = 0;
+  std::atomic_signal_fence(std::memory_order_release);
```

Comment:
nit: This is not necessary, this only tells the compiler to avoid re-ordering writes to the `s_sigchld_received`  variable after the fence. However:

- This is already enforced at each read or write of this variable due to the `volatile` qualifier when defining this variable.

- The next function call to `ppoll()` will act as a memory barrier as well.